src/org/omegat/core/data/RealProject.java

```diff
@@ -334,51 +336,77 @@ public synchronized void loadProject(boolean onlineMode) {
-                    remoteRepositoryProvider.switchAllToLatest();
-                } catch (IRemoteRepository2.NetworkException e) {
-                    Log.logErrorRB("TEAM_NETWORK_ERROR", e.getCause());
-                    setOfflineMode();
+                if (remoteRepositoryProvider != null) {
+                    try {
+                        tmxPrepared = null;
+                        glossaryPrepared = null;
+                        remoteRepositoryProvider.switchAllToLatest();
+                    } catch (IRemoteRepository2.NetworkException e) {
+                        Log.logErrorRB("TEAM_NETWORK_ERROR", e.getCause());
+                        setOfflineMode();
+                    }
+                    remoteRepositoryProvider.copyFilesFromReposToProject("");
+
+                    // After adding filters.xml and segmentation.conf, we must
+                    // reload them again
+                    config.loadProjectFilters();
+                    config.loadProjectSRX();
                 }
-                remoteRepositoryProvider.copyFilesFromReposToProject("");
 
-                // After adding filters.xml and segmentation.conf, we must
-                // reload them again
-                config.loadProjectFilters();
-                config.loadProjectSRX();
+                loadFilterSettings();
+                loadSegmentationSettings();
+                loadTranslations(); // load project_save.tmx
+                loadSourceFiles();
+            } catch (XMLStreamException e) {
+                // omegat.project file corrupted.
+                Log.logErrorRB(e, "TF_LOAD_ERROR_PARSER_EXCEPTION");
+                Core.getMainWindow().displayErrorRB(e, "TF_LOAD_ERROR_PARSER_EXCEPTION");
+                if (!loaded) {
+                    unlockProject();
+                    ;
+                }
+                Log.logInfoRB("LOG_DATAENGINE_LOAD_END");
+                return;
             }
 
-            loadFilterSettings();
-            loadSegmentationSettings();
-            loadTranslations(); // load projectsave.tmx
-            loadSourceFiles();
-
-            // This MUST happen after calling loadTranslations()
-            if (remoteRepositoryProvider != null && isOnlineMode) {
-                Core.getMainWindow().showStatusMessageRB("TEAM_REBASE_AND_COMMIT");
-                rebaseAndCommitProject(true);
+            try {
+                // This MUST happen after calling loadTranslations()
+                if (remoteRepositoryProvider != null && isOnlineMode) {
+                    Core.getMainWindow().showStatusMessageRB("TEAM_REBASE_AND_COMMIT");
+                    // rebaseAndCommitProject can raise IOException,
+                    // XMLStreamException
+                    rebaseAndCommitProject(true);
+                }
+            } catch (XMLStreamException e) {
+                // omegat.project file corrupted.
+                Log.logErrorRB(e, "TF_LOAD_ERROR_PARSER_EXCEPTION");
```

Comment:
this can be under `if` block. When team project sync failed, show error message.